js/source/vector_tile_worker_source.js

```diff
@@ -48,20 +49,27 @@ VectorTileWorkerSource.prototype = {
         if (!this.loading[source])
             this.loading[source] = {};
 
-        var tile = this.loading[source][uid] = new WorkerTile(params);
-        tile.abort = this.loadVectorData(params, done.bind(this));
+        var workerTile = this.loading[source][uid] = new WorkerTile(params);
+        workerTile.abort = this.loadVectorData(params, done.bind(this));
 
-        function done(err, data) {
+        function done(err, vectorTile) {
             delete this.loading[source][uid];
 
             if (err) return callback(err);
-            if (!data) return callback(null, null);
+            if (!vectorTile) return callback(null, null);
 
-            tile.data = data.tile;
-            tile.parse(tile.data, this.styleLayers.getLayerFamilies(), this.actor, data.rawTileData, callback);
+            workerTile.vectorTile = vectorTile;
+            workerTile.parse(vectorTile, this.styleLayers.getLayerFamilies(), this.actor, function (err, result, transferrables) {
+                if (err) return callback(err);
+
+                // Not transferring rawTileData because the worker needs to retain its copy.
+                callback(null,
+                    util.extend({rawTileData: vectorTile.rawData}, result),
```

Comment:
Or perhaps more to the point -- `WorkerTile#parse` is used for both loading and reloading, and part of the purpose of this PR is to stop sending `rawTileData` for reloading. So if `WorkerTile#parse` were to set it, it would also have to know if it was being used for loading or reloading.